src/pybind/mgr/cephadm/serve.py

```diff
@@ -1073,6 +1074,52 @@ def _write_client_files(self,
         if updated_files:
             self.mgr.cache.save_host(host)
 
+    def _write_all_tuned_profiles(self) -> None:
+        host_profile_mapping: Dict[str, List[Dict[str, str]]] = {}
+        for host in self.mgr.cache.get_hosts():
+            host_profile_mapping[host] = []
+
+        for profile in self.mgr.tuned_profiles.list_profiles():
+            p_str = f'# created by cephadm\n# tuned profile "{profile.profile_name}"\n\n'
+            for k, v in profile.settings.items():
+                p_str += f'{k} = {v}\n'
+            ha = HostAssignment(
+                spec=ServiceSpec(
+                    'crash', placement=profile.placement),
+                hosts=self.mgr.cache.get_schedulable_hosts(),
+                unreachable_hosts=self.mgr.cache.get_unreachable_hosts(),
+                daemons=[],
+                networks=self.mgr.cache.networks,
+            )
+            all_slots, _, _ = ha.place()
+            for host in {s.hostname for s in all_slots}:
+                host_profile_mapping[host].append({profile.profile_name: p_str})
+
+        for host, profiles in host_profile_mapping.items():
+            self._remove_stray_tuned_profiles(host, profiles)
+            self._write_tuned_profiles(host, profiles)
+
+    def _remove_stray_tuned_profiles(self, host: str, profiles: List[Dict[str, str]]) -> None:
+        cmd = ['ls', '/etc/sysctl.d']
```

Comment:
```suggestion
        cmd = ['ls', SYSCRL_DIR]
```
create a constant for this dir?

Reply:
done